pkg/api/types.go

```diff
@@ -826,6 +826,11 @@ type TestStepConfiguration struct {
 	// RestrictNetworkAccess restricts network access to RedHat intranet.
 	RestrictNetworkAccess *bool `json:"restrict_network_access,omitempty"`
 
+	// ShardCount describes the number of jobs that should be generated as shards for this test
+	// Each generated job will be a duplication, but contain a suffix and the necessary SHARD_ARGS will be passed to the steps
+	// Only applicable to presubmits and periodics
+	ShardCount *int `json:"shard_count,omitempty"`
```

Comment:
I don't think that this is the right place. The shard count, is a prowgen configuration. Since ci-operator is not related here, it should be better to allow the user to control the extra prowgen behavior for the job generation.
Perhaps in the `.config.prowgen`:
```yaml
jobs:
  test-job:
     shard_count: 2
     other_config: ...
```

Reply:
I was expecting you would say that, but many of the features that we have added recently have been implemented in the same manner. [Retry](https://github.com/openshift/ci-tools/blob/1960063e9e900f4ca3665742fbbad18529b7bf97/pkg/api/types.go#L801), for instance. I considered adding this to the prowgen config, but the implementation would be considerably more complex, and more difficult to use.

Reply:
I think that if the prowgen config were redesigned to be more "configurer"-friendly, it would be a better selling point to move this there. That is definitely a larger effort though. IMO, leaving it here until that takes place is the best move.